### PR TITLE
Move test suite offline using pytest-httpbin

### LINKED ISSUE #2184: Port test suite to pytest-httpbin
https://github.com/kevin1024/pytest-httpbin

### PR DESCRIPTION
**Pull request by Lukasa**
Resolves #2184, and should allow us to move back to Travis for our testing.

### CODE REVIEW

**Comment by Lukasa** [comment]
There _ought_ to be a lot of improvements to doing this. Tests should be a lot closer to reproducible, and the runtime should improve greatly. This is still not really how I want this test suite to look, but it's a _lot_ closer.

File: test_requests.py [hunk 1]
```diff
@@ -431,22 +448,7 @@ def test_DIGESTAUTH_QUOTES_QOP_VALUE(self):
         r = requests.get(url, auth=auth)
         assert '"auth"' in r.request.headers['Authorization']
 
-    def test_DIGESTAUTH_THREADED(self):
```
Comment (sigmavirus24) [inline test_requests.py:434]: Could we add a `# TODO(...): Add threaded HTTPDigestAuth test`?

**Comment by sigmavirus24** [comment]
One request, then I'll press the big green button. :)

Alternatively, we can add a bug to track the work of adding that test.

**Comment by Lukasa** [comment]
I think it's better to add a bug than a note in the code: we'll just lose the note. ;)

**Comment by sigmavirus24** [comment]
Thanks @Lukasa!

**Pull request merged**